src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/HandlerOrchestrator.cs

```diff
@@ -12,29 +13,31 @@ internal sealed class HandlerOrchestrator
         /// <summary>
         /// Class responsible for creation of chain of handlers and initiate the chain
         /// </summary>
-        private ConnectionHandlerContext _initialConnectionHandlerContext;
-
-        public HandlerOrchestrator(ConnectionHandlerContext initialConnectionHandlerContext)
+        private static readonly Lazy<HandlerOrchestrator> lazyInstance = new Lazy<HandlerOrchestrator>(() => new HandlerOrchestrator());
+        private HandlerOrchestrator()
         {
-            _initialConnectionHandlerContext = initialConnectionHandlerContext;
         }
 
+        public static HandlerOrchestrator Instance => lazyInstance.Value;
+
         /// <summary>
         /// Intiates the pre-defined chain of handlers
         /// </summary>
+        /// <param name="handlerContext"></param>
         /// <param name="isAsync"></param>
         /// <param name="ct"></param>
         /// <returns></returns>
-        public async ValueTask ProcessRequestAsync(bool isAsync, CancellationToken ct) 
+        public async ValueTask ProcessRequestAsync(ConnectionHandlerContext handlerContext, bool isAsync, CancellationToken ct) 
         {
-            //TODO: replace with a static line of creating the chain
             DataSourceParsingHandler dataSourceParsingHandler = new DataSourceParsingHandler();
             TransportCreationHandler transportCreationHandler = new TransportCreationHandler();
+            SSRPHandler ssrpHandler = new SSRPHandler();
             PreloginHandler preloginHandler = new PreloginHandler();
-            dataSourceParsingHandler.NextHandler = transportCreationHandler;
+            dataSourceParsingHandler.NextHandler = ssrpHandler;
+            ssrpHandler.NextHandler = transportCreationHandler;
             transportCreationHandler.NextHandler = preloginHandler;
-
-            await dataSourceParsingHandler.Handle(_initialConnectionHandlerContext, isAsync, ct).ConfigureAwait(false);
+            
+            await dataSourceParsingHandler.Handle(handlerContext, isAsync, ct).ConfigureAwait(false);
```

Comment:
Same here, we can return the result of Handle directly:
```suggestion
            return dataSourceParsingHandler.Handle(handlerContext, isAsync, ct);
```